### PR TITLE
Add some notes on shapes layer state.

### PR DESCRIPTION
**Pull request by perlman**
Some notes on shape state.
(I don't have direct push access to your branch).

### CODE REVIEW

File: docs/naps/4-async-slicing.md [hunk 1]
```diff
@@ -223,6 +223,28 @@ It's important to understand what state is currently used for slicing in napari.
     - `_max_points_thumbnail`: `int`, if more points than this in slice, randomly sample them
     - `_selected_view`: `list[int]`, intersection of `_selected_data` and `_indices_view`, could be a cached property
 
+- `Shapes`
+    - `_data_view`: `ShapeList`, container around shape data
+    - `_feature_table`: `_FeatureTable`, dataframe-like features table, scales to number of shapes
+
+	- `ShapeList`
+		- `_slice_key`: `List(int)`, current slice key
```
Comment (andy-sweet) [inline docs/naps/4-async-slicing.md]:
```suggestion
		- `_slice_key`: `list[int]`, current slice key
```

For consistency with the notes above.

File: docs/naps/4-async-slicing.md [hunk 2]
```diff
@@ -223,6 +223,28 @@ It's important to understand what state is currently used for slicing in napari.
...
+		- `_mesh`: `Mesh`, container to store meshes from shapes
+		- `shapes`: `List(Shape)`, list of underlying shapes
```
Comment (andy-sweet) [inline docs/naps/4-async-slicing.md]:
```suggestion
		- `shapes`: `list[Shape]`, list of underlying shapes
```

File: docs/naps/4-async-slicing.md [hunk 3]
```diff
@@ -223,6 +223,28 @@ It's important to understand what state is currently used for slicing in napari.
     - `_max_points_thumbnail`: `int`, if more points than this in slice, randomly sample them
     - `_selected_view`: `list[int]`, intersection of `_selected_data` and `_indices_view`, could be a cached property
 
+- `Shapes`
+    - `_data_view`: `ShapeList`, container around shape data
+    - `_feature_table`: `_FeatureTable`, dataframe-like features table, scales to number of shapes
+
+	- `ShapeList`
```
Comment (andy-sweet) [inline docs/naps/4-async-slicing.md]:
```suggestion
    - `ShapeList`
```

File: docs/naps/4-async-slicing.md [hunk 4]
```diff
@@ -223,6 +223,28 @@ It's important to understand what state is currently used for slicing in napari.
...
+		- `_slice_key`: `List(int)`, current slice key
+		- `_mesh`: `Mesh`, container to store meshes from shapes
+		- `shapes`: `List(Shape)`, list of underlying shapes
+		- `displayed_vertices`, `List`
+		- `displayed_index`, `List`
+
+	- `Shape` (and subclasses... `PolygonBase`, `Polygon`, etc.)
+		- `slice_key`: `List(int)`, min/max of non-displayed dimensions
```
Comment (andy-sweet) [inline docs/naps/4-async-slicing.md]:
```suggestion
		- `slice_key`: `list[int]`, min/max of non-displayed dimensions
```

File: docs/naps/4-async-slicing.md [hunk 5]
```diff
@@ -223,6 +223,28 @@ It's important to understand what state is currently used for slicing in napari.
     - `_max_points_thumbnail`: `int`, if more points than this in slice, randomly sample them
     - `_selected_view`: `list[int]`, intersection of `_selected_data` and `_indices_view`, could be a cached property
 
+- `Shapes`
+    - `_data_view`: `ShapeList`, container around shape data
+    - `_feature_table`: `_FeatureTable`, dataframe-like features table, scales to number of shapes
```
Comment (andy-sweet) [inline docs/naps/4-async-slicing.md]: If this isn't directly related to slicing, might be best to drop it to be more concise. Or put it in a more detailed notes section.

File: docs/naps/4-async-slicing.md [hunk 6]
```diff
@@ -223,6 +223,28 @@ It's important to understand what state is currently used for slicing in napari.
...
+		- `shapes`: `List(Shape)`, list of underlying shapes
+		- `displayed_vertices`, `List`
+		- `displayed_index`, `List`
+
+	- `Shape` (and subclasses... `PolygonBase`, `Polygon`, etc.)
+		- `slice_key`: `List(int)`, min/max of non-displayed dimensions
+
+	- `Mesh`
+	    - Data to be shown
+			- `displayed_triangles`: `Array`, Triamgles to be drawn
```
Comment (andy-sweet) [inline docs/naps/4-async-slicing.md]:
```suggestion
			- `displayed_triangles`: `Array`, triangles to be drawn
```

Also fix the indentation here.

File: docs/naps/4-async-slicing.md [hunk 7]
```diff
@@ -223,6 +223,28 @@ It's important to understand what state is currently used for slicing in napari.
...
+- `Shapes`
+    - `_data_view`: `ShapeList`, container around shape data
+    - `_feature_table`: `_FeatureTable`, dataframe-like features table, scales to number of shapes
+
+	- `ShapeList`
+		- `_slice_key`: `List(int)`, current slice key
+		- `_mesh`: `Mesh`, container to store meshes from shapes
+		- `shapes`: `List(Shape)`, list of underlying shapes
+		- `displayed_vertices`, `List`
+		- `displayed_index`, `List`
```
Comment (andy-sweet) [inline docs/naps/4-async-slicing.md]: Is this similar to `_indices_view` in `Points`? I.e. the indices of the shapes that are in the current slice? What type of elements are in the list? ints?

File: docs/naps/4-async-slicing.md [hunk 8]
```diff
@@ -223,6 +223,28 @@ It's important to understand what state is currently used for slicing in napari.
...
 
+- `Shapes`
+    - `_data_view`: `ShapeList`, container around shape data
+    - `_feature_table`: `_FeatureTable`, dataframe-like features table, scales to number of shapes
+
+	- `ShapeList`
+		- `_slice_key`: `List(int)`, current slice key
+		- `_mesh`: `Mesh`, container to store meshes from shapes
+		- `shapes`: `List(Shape)`, list of underlying shapes
+		- `displayed_vertices`, `List`
```
Comment (andy-sweet) [inline docs/naps/4-async-slicing.md]: Is this something like `self._mesh.vertices[self.displayed_index]? Would be good to add a short description.

**Review by andy-sweet — CHANGES_REQUESTED** [review]
Thanks for digging @perlman - this is super useful info.

I had a few nits mostly around type annotation style and indentation, but nothing major. Once you address those or push back, I'll merge this.

**Comment by andy-sweet** [comment]
You should also add yourself as an author to the doc at the top. Email is optional.

**Comment by perlman** [comment]
@andy-sweet Ok, should be a bit neater!

**Review by andy-sweet — APPROVED** [review]
Thanks for the updates! Looks good.

**Pull request merged**